In [3]:
import pandas as pd
# 두 단계 밖으로 나갔다가 data 폴더로 들어가는 경로
df = pd.read_csv('../../data/customer_hm.csv') 
df.head()

customer_id  FN  Active  \
0  00000dbacae5abe5e23885899a1fa44253a17956c6d1c3...   0       0   
1  0000423b00ade91418cceaf3b26c6af3dd342b51fd051e...   0       0   
2  000058a12d5b43e67d225668fa1f8d618c13dc232df0ca...   0       0   
3  00005ca1c9ed5f5146b52ac8639a40ca9d57aeff4d1bd2...   0       0   
4  00006413d8573cd20ed7128e53b7b13819fe5cfc2d801f...   1       1   

  club_member_status fashion_news_frequency  age  
0             ACTIVE                   NONE   49  
1             ACTIVE                   NONE   25  
2             ACTIVE                   NONE   24  
3             ACTIVE                   NONE   54  
4             ACTIVE              Regularly   52

In [6]:
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 6 columns):
 #   Column                  Non-Null Count    Dtype
---  ------                  --------------    -----
 0   customer_id             1048575 non-null  str  
 1   FN                      1048575 non-null  int64
 2   Active                  1048575 non-null  int64
 3   club_member_status      1048575 non-null  str  
 4   fashion_news_frequency  1048574 non-null  str  
 5   age                     1048575 non-null  int64
dtypes: int64(3), str(3)
memory usage: 48.0 MB
None


In [7]:
print(df.describe())

                 FN        Active           age
count  1.048575e+06  1.048575e+06  1.048575e+06
mean   3.555397e-01  3.461326e-01  3.636919e+01
std    4.786767e-01  4.757363e-01  1.430899e+01
min    0.000000e+00  0.000000e+00  1.600000e+01
25%    0.000000e+00  0.000000e+00  2.400000e+01
50%    0.000000e+00  0.000000e+00  3.200000e+01
75%    1.000000e+00  1.000000e+00  4.900000e+01
max    1.000000e+00  1.000000e+00  9.900000e+01


In [8]:
print(df['age'].value_counts())

age
21    52237
24    43377
20    42891
23    42615
25    42425
      ...  
97        4
95        4
93        3
94        3
96        2
Name: count, Length: 84, dtype: int64


In [9]:
# 각 상태별로 고객이 몇 명인지 확인
print(df['club_member_status'].value_counts())

club_member_status
ACTIVE        982635
PRE-CREATE     65581
LEFT CLUB        359
Name: count, dtype: int64


각 상태별 0,1,2로 매핑 가능
머신러닝에는 효율적이나, 시각화에는 비효율적

각 상태별 고객 수 현황
ACTIVE (활성): 982,635명 (약 93.7%)
PRE-CREATE (신규): 65,581명 (약 6.2%)
LEFT CLUB (탈퇴): 359명 (약 0.03%)

주요 인사이트
전체 고객의 93% 이상이 활성 멤버십 상태, 매우 높은 편

In [10]:
print(df.isnull().sum())

customer_id               0
FN                        0
Active                    0
club_member_status        0
fashion_news_frequency    1
age                       0
dtype: int64


결측치가 거의 없는 깔끔한 데이터
fashion_news_frequency의 결측값은 1개이니까 삭제하거나 최빈값인 NONE으로 채우려고 했는데 
FN상태 확인해보고 채우기로

In [14]:
print(df['FN'].value_counts())

FN
0    675765
1    372810
Name: count, dtype: int64


In [13]:
print(df['fashion_news_frequency'].value_counts())

fashion_news_frequency
NONE         674698
Regularly    373218
Monthly         658
Name: count, dtype: int64


In [ ]:
# 각 패션 뉴스 수신 빈도의 비율
print(df['fashion_news_frequency'].value_counts(normalize=True)*100)

fashion_news_frequency
NONE         64.344338
Regularly    35.592910
Monthly       0.062752
Name: proportion, dtype: float64


패션 뉴스 구독 현황
NONE(구독 안함): 674,698명
Regularly(정기적): 373,218명
Monthly(달마다): 658명

주요 인사이트
패션 뉴스 구독(마케팅 활성화)가 필요한 64%의 잠재 고객층이 존재

향후 가능 분석
구독 여부(FN)와 연령대(age)의 상관관계를 분석하여 타겟 마케팅 전략 수립 가능

In [ ]:
#  구독하는 사람(1)과 안 하는 사람(0)의 평균 나이는 다를까?
print(df.groupby('FN')['age'].mean())
# 구독하는 사람(1)과 안 하는 사람(0)의 평균 나이는 크게 다르지 않다

FN
0    35.999780
1    37.038805
Name: age, dtype: float64


In [ ]:
# 나이대를 10대, 20대... 순으로 나누기
df['age_group'] = (df['age'] // 10) * 10

# 나이대별로 패션 뉴스 구독 비율 확인
fn_rate_by_age = df.groupby('age_group')['FN'].mean()
print(fn_rate_by_age)

age_group
10    0.372050
20    0.344468
30    0.321967
40    0.362266
50    0.376756
60    0.421299
70    0.443814
80    0.400930
90    0.277778
Name: FN, dtype: float64
